newIDE/app/src/AssetStore/ExtensionStore/ExtensionsSearchDialog.js

```diff
@@ -73,19 +109,7 @@ export default function ExtensionsSearchDialog({
                 key="import"
                 label={<Trans>Import extension</Trans>}
                 onClick={() => {
-                  (async () => {
-                    setIsInstalling(true);
-                    const wasExtensionImported = await importExtension(
-                      i18n,
-                      eventsFunctionsExtensionsState,
-                      project
-                    );
-                    if (wasExtensionImported && onExtensionInstalled)
-                      onExtensionInstalled();
-
-                    setExtensionWasInstalled(wasExtensionImported);
-                    setIsInstalling(false);
-                  })();
+                  installOrImportExtension(i18n);
```

Comment:
Was there an issue in the existing logic, or is this just refactoring?

Reply:
Just refactoring to have logic only in one place